tools/resourcedocsgen/cmd/metadata_test.go

```diff
@@ -67,41 +69,65 @@ func TestMetadataNativeProvider(t *testing.T) {
 func TestMetadataComponentProvider(t *testing.T) {
 	t.Parallel()
 	testMetadata(t, testMetadataArgs{
-		providerName: "aws-apigateway",
-		version:      "v2.6.1",
-		schemaFile:   "schema.yaml",
+		repoSlug:   "pulumi/pulumi-aws-apigateway",
+		version:    "v2.6.1",
+		schemaFile: "schema.yaml",
 	})
 }
 
 type testMetadataArgs struct {
-	providerName, version       string
+	repoSlug, version           string
 	schemaFile                  string
 	schemaFileURL, indexFileURL string
 	assert                      func(t *testing.T, metadataDir, pacakgeDocsDir string)
+	assertOptions               []util.OptionAssertDirsEqual
+	errorContains               string
 }
 
-func defaultAssert(t *testing.T, metadataDir, pacakgeDocsDir string) {
-	t.Run("metadata", func(t *testing.T) { t.Parallel(); util.AssertDirEqual(t, metadataDir) })
-	t.Run("index", func(t *testing.T) { t.Parallel(); util.AssertDirEqual(t, pacakgeDocsDir) })
+func defaultAssert(t *testing.T, metadataDir, pacakgeDocsDir string, opts ...util.OptionAssertDirsEqual) {
+	t.Run("metadata", func(t *testing.T) { t.Parallel(); util.AssertDirEqual(t, metadataDir, opts...) })
+	t.Run("index", func(t *testing.T) { t.Parallel(); util.AssertDirEqual(t, pacakgeDocsDir, opts...) })
 }
 
 func testMetadata(t *testing.T, args testMetadataArgs) {
 	cmd := PackageMetadataCmd()
 	metadataDir := t.TempDir()
 	pacakgeDocsDir := t.TempDir()
 	cmd.SetArgs([]string{
-		"--repoSlug", "pulumi/pulumi-" + args.providerName,
+		"--repoSlug", args.repoSlug,
 		"--schemaFile", args.schemaFile,
 		"--schemaFileURL", args.schemaFileURL,
 		"--indexFileURL", args.indexFileURL,
 		"--version", args.version,
 		"--metadataDir", metadataDir,
 		"--packageDocsDir", pacakgeDocsDir,
 	})
+
+	// Capture output into the test
+	var stdout bytes.Buffer
+	var stderr bytes.Buffer
+	cmd.SetOut(&stdout)
+	cmd.SetErr(&stderr)
+	defer func() {
+		if stdout.Len() > 0 {
+			t.Log(stdout.String())
+		}
+		if stderr.Len() > 0 {
+			t.Log(stderr.String())
+		}
+	}()
+
+	if args.errorContains != "" {
+		assert.ErrorContains(t, cmd.Execute(), args.errorContains)
+		assert.Nil(t, args.assert,
+			"args.assert will not be called when args.errorContains is non-empty, so it should not be set")
```

Comment:
Because `args.errorContains != ""` means that we are asserting that there is an error (and what the error contains). I added a comment on the `errorContains` field that explain.